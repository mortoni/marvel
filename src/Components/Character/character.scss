.character {
    margin-bottom: 3rem;
    cursor: pointer;
    
    .thumbnail {
        img {
            width: 100%;
        }

        .caption {

        }
    }
}
