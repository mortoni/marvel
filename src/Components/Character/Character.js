import React from 'react';
import './character.scss';
import PropTypes from 'prop-types';

const Character = (props) => {
    const { thumbnail, name, onOpenModal } = props;
    const uri = `${thumbnail.path}/portrait_fantastic.${thumbnail.extension}`;
    return (
        <div
            className="col-sm-6 col-md-2 character"
            onClick={ () => onOpenModal(props) }
        >
            <div className="thumbnail">
                <img
                    src={ uri }
                    alt="Marvel character"/>
                <div className="caption">
                    <h5>{ name }</h5>
                </div>
            </div>
        </div>
    );
}

Character.propTypes = {
    thumbnail: PropTypes.object.isRequired,
    name: PropTypes.string.isRequired,
    onOpenModal: PropTypes.func.isRequired,
};

export default Character
