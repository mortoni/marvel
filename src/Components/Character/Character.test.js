import React from 'react';
import Character from './Character';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import Enzyme, { render, shallow } from 'enzyme';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });

describe('Testing App component', () => {
    const props = {
        thumbnail: {
            extension: 'jpg',
            path: 'http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available'
        },
        name: 'Wolver-dok',
        onOpenModal: () => {}
    };

    it('renders correctly', () => {
        const wrapper = shallow(<Character {...props}/>);
        expect(wrapper).toBeDefined();
    });

    it('renders as expected', () => {
        const wrapper = shallow(<Character {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
