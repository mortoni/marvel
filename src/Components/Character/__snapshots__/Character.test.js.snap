// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Character
    name="Wolver-dok"
    onOpenModal={[Function]}
    thumbnail={
      Object {
        "extension": "jpg",
        "path": "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="thumbnail"
      >
        <img
          alt="Marvel character"
          src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg"
        />
        <div
          className="caption"
        >
          <h5>
            Wolver-dok
          </h5>
        </div>
      </div>,
      "className": "col-sm-6 col-md-2 character",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Marvel character"
            src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg"
          />,
          <div
            className="caption"
          >
            <h5>
              Wolver-dok
            </h5>
          </div>,
        ],
        "className": "thumbnail",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Marvel character",
            "src": "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h5>
              Wolver-dok
            </h5>,
            "className": "caption",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Wolver-dok",
            },
            "ref": null,
            "rendered": "Wolver-dok",
            "type": "h5",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="thumbnail"
        >
          <img
            alt="Marvel character"
            src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg"
          />
          <div
            className="caption"
          >
            <h5>
              Wolver-dok
            </h5>
          </div>
        </div>,
        "className": "col-sm-6 col-md-2 character",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Marvel character"
              src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg"
            />,
            <div
              className="caption"
            >
              <h5>
                Wolver-dok
              </h5>
            </div>,
          ],
          "className": "thumbnail",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Marvel character",
              "src": "http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available/portrait_fantastic.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h5>
                Wolver-dok
              </h5>,
              "className": "caption",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Wolver-dok",
              },
              "ref": null,
              "rendered": "Wolver-dok",
              "type": "h5",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
